### DISCUSSION TITLE
How to make black scatter points in CIE Diagram?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#742] [Q&A] [answered]
Hi Colour Team,

In the CIE Diagram of one image, the scatter points colors are based on its distance to the white point, right?
![image](https://user-images.githubusercontent.com/16201681/80296334-105ac280-8748-11ea-85f7-5b5a31e7f9d8.png)

But now I want to make all of them just pure BLACK points.
Is there any option to do that?

Thanks

**Comment by thomasmansencal** [comment, marked as answer]
Hi @hminle,

The points are actually using the colours of the array you pass in, you can, however, change them entirely using the `scatter_kwargs` arguments: https://github.com/colour-science/colour/blob/817b1aa9a7456cf4c37e93adbf374b950d4cb0cf/colour/plotting/models.py#L663

It passes the `dict` directly to the [matplotlib.pyplot.scatter](https://matplotlib.org/3.2.1/api/_as_gen/matplotlib.pyplot.scatter.html) definition.

**Comment by thomasmansencal** [comment]
By the way, if you use `colour.plotting.colour_style()` you will get default plotting settings that are maybe better for you!

**Comment by hminle** [comment]
> colour.plotting.colour_style()

Thank @KelSolaar ,
But none of that is what I want.
I want to make all scatter points have the black color.
Now each point has different gradient colors.

Thanks

**Comment by thomasmansencal** [comment]
Did you try something like that: https://colab.research.google.com/drive/1NRcdXSCshivkwoU2nieCvC3y14fx1X4X#scrollTo=pPmX3lofHFxF&line=9&uniqifier=1

![image](https://user-images.githubusercontent.com/99779/80296895-b2bf8800-87d2-11ea-9753-8b11690bd90b.png)

**Comment by hminle** [comment]
Hi @KelSolaar Ke
Yes, I did try
It's still the same
![image](https://user-images.githubusercontent.com/16201681/80297073-089e1c80-874e-11ea-807a-515784a1fef5.png)

The scatter points still do not have consistent black color

**Comment by thomasmansencal** [comment]
Mmmh, this is odd! Can you please let me know the output of that call: `colour.utilities.describe_environment();` and also paste the whole invocation for the plotting call.

**Comment by hminle** [comment]
Hi @KelSolaar 
My command:
```
plotting.plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931(
    source,
    colourspace='sRGB',
    title="Input sRGB",
    scatter_kwargs={'c': 'k', 'marker': 'o'},
    show_pointer_gamut=False,
    show_whitepoints=False, 
    show_spectral_locus=False,
    standalone=True)
```
output of environment:
![image](https://user-images.githubusercontent.com/16201681/80297230-52d3cd80-874f-11ea-8331-c6193273f9aa.png)

Thanks

**Comment by thomasmansencal** [comment]
Trying to look at it, could be an issue with MPL.

**Comment by hminle** [comment]
> Trying to look at it, could be an issue with MPL.

You mean mismatch version?

**Comment by thomasmansencal** [comment]
Yup! Testing on Colab atm but having a hard time to get that particular version installed.

**Comment by thomasmansencal** [comment]
Oh, can you try `scatter_args` instead of `scatter_kwargs`? `scatter_kwargs` is for the latest `develop` branch! :)

**Comment by hminle** [comment]
> Oh, can you try `scatter_args` instead of `scatter_kwargs`? `scatter_kwargs` is for the latest `develop` branch! :)

I've just tried.
It's still the same.
Don't know why, but any scatter arguments seems do not have any effect on the Diagram.

**Comment by thomasmansencal** [comment]
Sorry `scatter_parameters`, the deprecation code is incorrect in `develop` :(

**Comment by hminle** [comment]
> scatter_parameters

Great! It works now
Thank you very much.
:)

**Comment by thomasmansencal** [comment]
Pfiouuu! Sweet! :) Note that `scatter_parameters` will effectively be renamed to `scatter_kwargs` in 0.3.16 but you will get a warning and `scatter_parameters` will still work.

**Comment by hminle** [comment]
Yeah, so I just need to upgrade colour.
Thank guys :)

**Comment by thomasmansencal** [comment]
It is not yet out but slowly working on that! Don't worry about it for now ;)
